app/src/main/java/ch/epfl/culturequest/social/Profile.java

```diff
@@ -1,40 +1,80 @@
 package ch.epfl.culturequest.social;
 
-import android.net.Uri;
+import androidx.annotation.NonNull;
 
 import com.google.firebase.auth.FirebaseAuth;
 import com.google.firebase.auth.FirebaseUser;
 
+import java.util.HashMap;
+import java.util.List;
 import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import java.util.Observable;
+import java.util.Observer;
+
+import ch.epfl.culturequest.database.Database;
 
 /**
  * Creates a profile for users
  */
-public class Profile {
+public class Profile extends Observable{
+
+    private String uid, name, username, email, phoneNumber;
+    private String profilePicture;
+
+
+    private List<Image> images;
+
 
-    private final String uid, name, username, email, phoneNumber;
-    private Uri profilePicture;
 
     /**
      * Creates a Profile for a user.
-     *
+     * <p>
      * All Preconditions are checked in other classes. e.g., We create a profile
      * only when the user has correctly signed in and has correctly set they're username.
      * So user will normally never be null, and there is no need for sanitization of username as
      * it's already been dealt with
      *
-     * @param user current user signed in
-     * @param username username of user
+     * @param username       username of user
      * @param profilePicture Profile picture. Can be set to null
      */
-    public Profile(FirebaseUser user, String username, Uri profilePicture) {
-        user = Objects.requireNonNull(FirebaseAuth.getInstance().getCurrentUser());
+    public Profile(String username, String profilePicture)  {
+        FirebaseUser user = Objects.requireNonNull(FirebaseAuth.getInstance().getCurrentUser());
         this.username = username;
         this.uid = user.getUid();
         this.name = user.getDisplayName();
         this.email = user.getEmail();
         this.phoneNumber = user.getPhoneNumber();
         this.profilePicture = profilePicture;
+        this.images = null;
+    }
+
+    public Profile(String uid, String name, String username, String email, String phoneNumber, String profilePicture, List<Image> images) {
```

Comment:
why is the profilePicture a string? isnt it better/easier to have it as a Uri?

Reply:
I had to choose a string because it is way easier with firebase (and in any case the conversion to a URI is trivial)